### ISSUE TITLE
OpenAPI filter (OASFilter) doesn't work

### ISSUE THREAD

**Issue by esvechnikov** [OpenNTF/org.openntf.xsp.jakartaee#504]
The task is to add externalDocs to openapi. 
I tried to do it through the configuration (microprofile-config.properties).
```properties
mp.openapi.extensions.smallrye.externalDocs.description=
mp.openapi.extensions.smallrye.externalDocs.url=
```
But the information does not end up in the openapi file.

I tried to create a filter, but the filter does not load.
```java
package com.example.filter;
public class ExampleOASFilter implements OASFilter {}
```
```properties
mp.openapi.filter=com.example.filter.ExampleOASFilter
```

How to override output in openapi?

**Comment by esvechnikov** [comment]
I connected OASFilter for this and registered it in the microprofile-config.properties file
```ini
mp.openapi.filter=com.example.filter.ExampleOASFilter
```

Now when I use the filter to override Info
```java
@Provider
public class APIOASFilter implements OASFilter {

	@Override
	public void filterOpenAPI(OpenAPI openAPI) {
		openAPI.setInfo(OASFactory
			.createInfo()
				.title("API")
				.description("API")
				.contact(OASFactory
					.createContact()
						.name("user")
						.email("user@mail.com"))
						.version("1.0.0"));		
	}

}
```

I get an error
```JSON
{
    "message": "java.lang.IllegalStateException: No OASFactoryResolver implementation found!",
    "stackTrace": [
        [
            "java.lang.IllegalStateException: No OASFactoryResolver implementation found!",
            "  at org.eclipse.microprofile.openapi.spi.OASFactoryResolver.instance(OASFactoryResolver.java:82)",
            "  at org.eclipse.microprofile.openapi.OASFactory.createObject(OASFactory.java:96)",
            "  at org.eclipse.microprofile.openapi.OASFactory.createInfo(OASFactory.java:195)",
            "  at com.example.microprofile.APIOASFilter.filterOpenAPI(APIOASFilter.java:21)",
            "  at io.smallrye.openapi.api.util.FilterUtil.filter(FilterUtil.java:68)",
            "  at io.smallrye.openapi.api.util.FilterUtil.applyFilter(FilterUtil.java:51)",
            "  at io.smallrye.openapi.api.OpenApiDocument.filterModel(OpenApiDocument.java:177)",
            "  at io.smallrye.openapi.api.OpenApiDocument.initialize(OpenApiDocument.java:134)",
            "  at io.smallrye.openapi.runtime.OpenApiProcessor.bootstrap(OpenApiProcessor.java:79)",
            "  at io.smallrye.openapi.runtime.OpenApiProcessor.bootstrap(OpenApiProcessor.java:52)",
            "  at org.openntf.xsp.microprofile.openapi.AbstractOpenAPIResource.buildOpenAPI(AbstractOpenAPIResource.java:81)",
            "  at org.openntf.xsp.microprofile.openapi.OpenAPIResource.get(OpenAPIResource.java:40)",
            "  at sun.reflect.GeneratedMethodAccessor507.invoke(Unknown Source)",
            "  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)",
            "  at java.lang.reflect.Method.invoke(Method.java:508)",
            "  at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)",
            "  at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)",
            "  at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:660)",
            "  at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:524)",
            "  at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:474)",
            "  at org.jboss.resteasy.core.ResourceMethodInvoker$$Lambda$325/0x000000001e7dcc90.get(Unknown Source)",
            "  at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)",
            "  at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:476)",
            "  at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:434)",
            "  at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:408)",
            "  at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:69)",
            "  at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)",
            "  at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)",
            "  at org.jboss.resteasy.core.SynchronousDispatcher$$Lambda$313/0x0000000013c611f0.run(Unknown Source)",
            "  at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)",
            "  at org.jboss.resteasy.core.SynchronousDispatcher$$Lambda$314/0x000000001e828290.get(Unknown Source)",
            "  at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)",
            "  at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)",
            "  at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)",
            "  at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:249)",
            "  at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:60)",
            "  at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:55)",
            "  at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:587)",
            "  at org.openntf.xsp.jaxrs.impl.FacesJAXRSServletContainer.doService(FacesJAXRSServletContainer.java:92)",
            "  at org.openntf.xsp.jakartaee.AbstractXspLifecycleServlet.service(AbstractXspLifecycleServlet.java:130)",
            "  at org.openntf.xsp.jaxrs.impl.FacesJAXRSServletContainer.service(FacesJAXRSServletContainer.java:66)",
            "  at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:587)",
            "  at org.openntf.xsp.jakartaee.servlet.NewHttpServletWrapper.service(NewHttpServletWrapper.java:112)",
            "  at com.ibm.designer.runtime.domino.adapter.ComponentModule.invokeServlet(ComponentModule.java:588)",
            "  at com.ibm.domino.xsp.module.nsf.NSFComponentModule.invokeServlet(NSFComponentModule.java:1335)",
            "  at com.ibm.designer.runtime.domino.adapter.ComponentModule$AdapterInvoker.invokeServlet(ComponentModule.java:865)",
            "  at com.ibm.designer.runtime.domino.adapter.ComponentModule$ServletInvoker.doService(ComponentModule.java:808)",
            "  at com.ibm.designer.runtime.domino.adapter.ComponentModule.doService(ComponentModule.java:577)",
            "  at com.ibm.domino.xsp.module.nsf.NSFComponentModule.doService(NSFComponentModule.java:1319)",
            "  at com.ibm.domino.xsp.module.nsf.NSFService.doServiceInternal(NSFService.java:662)",
            "  at com.ibm.domino.xsp.module.nsf.NSFService.doService(NSFService.java:482)",
            "  at com.ibm.designer.runtime.domino.adapter.LCDEnvironment.doService(LCDEnvironment.java:357)",
            "  at com.ibm.designer.runtime.domino.adapter.LCDEnvironment.service(LCDEnvironment.java:313)",
            "  at com.ibm.domino.xsp.bridge.http.engine.XspCmdManager.service(XspCmdManager.java:272)"
        ]
    ]
}
```

**Comment by jesse-gallagher** [comment]
This'll be fixed in 2.16.0 - turns out I needed to add a hook to set the default factory at init time.
